src/connect.js

```diff
@@ -82,29 +79,22 @@ export default (
       }
     }
 
-    _startPromises(props) {
-      const update = ({ key, data, error = false }) => {
-        if (!this._isMounted || this.hasTimedOut) return;
+    _update({ key, data, err }) {
+      if (!this._isMounted || this.hasTimedOut) return;
 
-        this.loadedCount += 1;
-        this.setState({
-          responses: {
-            ...this.state.responses,
-            [key]: data
-          },
-          errored: {
-            ...this.state.errored,
-            [key]: error
-          },
-          isLoading: this.promiseCount > this.loadedCount,
-        });
-      };
+      const responses = Object.assign({}, this.state.responses, { [key]: data });
+      const error = Object.assign({}, this.state.error, { [key]: err });
 
-      Object.entries(promises)
-        .forEach(([key, promise]) => promise(props)
-          .then((data) => update({ key, data }))
-          .catch((data) => update({ key, data, error: true }))
-      );
+      this.setState({ responses, error });
+    }
+
+    _startPromises(props) {
```

Comment:
Ah, saw that it's set to true when constructed. `_startPromises` should probably handle the toggle of the `isLoading` flag since it can realistically be called again after construction.